### ISSUE TITLE
Alter Android compatibilty of existing extensions to prepare for general availability of extensions on Fenix

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#21162]
We need to add a set of commands to run some data migrations to prepare for general availability of extensions on Fenix.

This is going to happen in multiple steps as we gradually expand the set of add-ons we're allowing in Beta/Nightly before fully opening up general availability.

For this step, which should land before Fenix 119 Beta is released:
- Force specific extensions as compatible with Firefox for Android 119.0a1 or higher from a csv file.
  - Should happen regardless of whether or not they were already marked as compatible with another version of Firefox for Android or not
  - Should drop `strict_compatibility` on relevant files
  - Should *not* alter compatibility of the current limited set of recommended/line extensions at any point, so that they continue to be installable by current Fenix release. If those end up in the csv file, they should be ignored.
- After that migration has been executed, force **all** versions in the database marked as compatible with Firefox for Android lower than 119.0a1 and not recommended/line for Android/All applications to be marked as compatible up to 68.* max instead
  - Should enable `strict_compatibility` on those files as well

**Comment by diox** [comment]
QA: when this lands, it won't do anything by default, you'll need to build a test csv with some add-on ids to test the various states described in the description, and share it with me so that I can run the commands. Then you should be able to check the compatibility data in devhub/admin tools (or API).

**Comment by ioanarusiczki** [comment]
@diox I prepared a list of IDs for AMO dev.

**Comment by diox** [comment]
For reference, using [ids.csv](https://github.com/mozilla/addons-server/files/12673807/ids.csv) on dev (which contains a mix of everything prepared by @ioanarusiczki):

Skipping add-on id 328839 because of its promoted group.
Skipping add-on id 491253 because of its promoted group.
Skipping add-on id 491817 because of its promoted group.
Skipping add-on id 497311 because of its promoted group.
Skipping add-on id 618623 because of its promoted group.
Skipping add-on id 620138 because of its promoted group.
Skipping add-on id 620247 because of its promoted group.
Skipping add-on id 620395 because of its promoted group.
Skipping add-on id 621034 because of its promoted group.
Skipping add-on id 621035 because of its promoted group.
Skipping add-on id 621036 because of its promoted group.
Skipping add-on id 625216 because of its promoted group.
Done forcing Android compatibility on 34 add-ons (12 skipped)
Done forcing max Android compatibility: Dropped compat on 494 versions, Updated compat for 6101 versions, Set strict compatibility on 6101 files

Done reindexing

**Comment by ioanarusiczki** [comment]
@diox 

Ids from csv:

1. Not compatible with Android Platform at submission: have now Android compatibility set to 119.0a1 and later
2. Compatible with both platforms at submission: Android compatibility is set to 119.0a1 when previously was 42.0 or 48.0 
3. Recommeded/line on Firefox (only Firefox compatible) are now compatible with Firefox for Android 119.0a1 and later
4. Recommeded/line on both platforms did not change their compatibility 
5. Lang packs or dictionaries have compatibility for Android 119.0a1 but they won't show up in the search results on Android

Except for the lang packs, strict_compat is disabled. 

If they were not on the list:
1. non-recommeded/line without Android compatibility did not change 
2. non-recommeded/line with Android compatibility are set for Android below 68*, strict_compatibility: enabled
3. recommeded for Firefox only are set for Android below 68*,  strict_compatibility: enabled

**Comment by diox** [comment]
That all looks good.

**Comment by diox** [comment]
On stage using [ids_stage.csv](https://github.com/mozilla/addons-server/files/12699468/ids_stage.csv):

Skipping add-on id 1005144 because of its promoted group.
Done forcing Android compatibility on 19 add-ons (1 skipped)
Done forcing max Android compatibility: Dropped compat on 708 versions, Updated compat for 4979 versions, Set strict compatibility on 4979 files
Reindexation done

**Comment by ioanarusiczki** [comment]
@diox 

I checked the noted ids on AMO stage. Looks good.

What I did not mention for AMO dev is that for addons with multiple versions which are compatible with both platforms -> they got Android compatibility below 68*.  
Examples: 1004477, 619792 or 620219 -> set on rejected/deleted versions too. The last version got 109.0a1 because it's in the list.

**Comment by ioanarusiczki** [comment]
@diox 

There's an ID on AMO stage which is an edge case - > 279910. 
Got recommeded on version 5.2.3 only on Firefox , yet was set for All Applications. 
And has compatibility set for Android 56.0a1 and later with Firefox 56.0a1 and later 

It is displayed on AMO stage - on Fenix release 117.1

**Comment by diox** [comment]
We discussed that specific add-on. It's because the migration command only looks at whether or not the add-on has a promoted addon instance for Android or All applications. Here it does have a promoted _addon_ instance for all applications, but a promoted _approval_ only for desktop. There are 3 reasons why this could happen:
- The Firefox Desktop approval was added manually through the admin
- The Firefox Android approval was deleted
- The Promoted Addon application was set to all after the approval had taken place

In all cases, we did intend for the add-on to be promoted on android, it's just not the case yet until a new version (or approval) is published.

**Comment by ioanarusiczki** [comment]
@diox Ids 279910 and 1002968 (found one more) in the same situation as described [above](https://github.com/mozilla/addons-server/issues/21162#issuecomment-1731570573)

**Comment by ioanarusiczki** [comment]
@diox 
On AMO stage - I'm attaching the results in my csv and marking this verified.
[Addon IDs on stage.csv](https://github.com/mozilla/addons-server/files/12715911/Addon.IDs.on.stage.csv)

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDSRV-488
